src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -544,22 +528,24 @@ export class MeetingManager implements AudioVideoObserver {
       this.selectedAudioInputDevice = null;
     } else {
       try {
-        await this.audioVideo?.chooseAudioInputDevice(receivedDevice);
+        await this.audioVideo?.startAudioInput(device);
         this.selectAudioInputDeviceError = null;
       } catch (error) {
         this.selectAudioInputDeviceError = error;
         console.error('Failed to choose audio input device.', error);
       }
 
-      let innerDevice = null;
+      let deviceId = null;
       if (isAudioTransformDevice(device)) {
-        innerDevice = await device.intrinsicDevice();
+        const innerDevice = await device.intrinsicDevice();
+        deviceId = DefaultDeviceController.getIntrinsicDeviceId(innerDevice);
       } else {
-        innerDevice = device;
+        const innerDevice = device;
+        if (innerDevice) {
+          deviceId = DefaultDeviceController.getIntrinsicDeviceId(innerDevice);
```

Comment:
The [`getIntrinsicDeviceId`](https://github.com/aws/amazon-chime-sdk-js/blob/main/src/devicecontroller/DefaultDeviceController.ts#L712) returns `undefined` for `null` `deviceId` so in that case should the `deviceId` be actually `undefined` or `null`? Not sure if that matters. @ltrung do you have any concerns?

Reply:
I refactored this code in my latest change (not in this PR yet) and `let deviceId = null;` is removed.
`deviceId` should be type of `string | string[] | undefined`.

Reply:
```tsx
let innerDevice: Device | null;

if (isVideoTransformDevice(device)) {
  innerDevice = await device.intrinsicDevice();
} else {
  innerDevice = device;
}

const deviceId = DefaultDeviceController.getIntrinsicDeviceId(innerDevice);
```